codegeex.el

```diff
@@ -125,17 +118,105 @@ buffer."
          (name-without-mode (replace-regexp-in-string "-mode" "" name)))
     name-without-mode))
 
+(defun codegeex--plist-get (plist &rest keys)
+  "Browse PLIST object for each KEYS and return the element
+where it stop.
+KEYS can be either numbers or properties symbols"
+  (let ((res json-result))
+    (dolist (key keys)
+      (cond
+       ((symbolp key)
+        (setq res (plist-get res key)))
+       ((numberp key)
+        (setq res (nth key res)))
+       (t
+        (error "Key '%s' format not supported" key))))
+    res))
+
+(defvar-local codegeex--completion-cache nil)
+(defvar-local codegeex--completion-idx 0)
+
 ;;;###autoload
-(defun codegeex-buffer-completion ()
-  "CodeGeeX buffer completion.
+(defun codegeex-complete ()
```

Comment:
`codegeex-buffer-completion` become:
- `codegeex-complete`, which callback is the overlay handler `codegeex-completion--show-completion`
- `codegeex-complete-at-point`, similar of previous `codegeex-buffer-completion`, which insert the first completion fetched